common/src/main/java/com/wynntils/features/redirects/ChatRedirectFeature.java

```diff
@@ -503,10 +505,13 @@ protected StyledText getNotification(Matcher matcher) {
     }
 
     private class LoginRedirector extends SimpleRedirector {
+        // Test suite: https://regexr.com/7jhti
+        private static final String RANK_STRING =
+                Arrays.stream(PlayerRank.values()).map(PlayerRank::getTag).collect(Collectors.joining());
         private static final Pattern FOREGROUND_PATTERN = Pattern.compile(
-                "^§.\\[(§.)+\\|?(§.)*(?<rank>[a-zA-Z+]+)(§.)+\\|?(§.)*\\] §.(?<name>.*)§. has just logged in!$");
-        private static final Pattern BACKGROUND_PATTERN = Pattern.compile(
-                "^(?:§8)?\\[(§.)+\\|?(§.)*(?<rank>[a-zA-Z+]+)(§.)+\\|?(§.)*\\] §7(?<name>.*)§8 has just logged in!$");
+                "^(?<rank>[" + RANK_STRING + "]) §#[0-9a-f]{8}(?<name>[\\w ]{1,20})§[0-9a-f] has just logged in!$");
```

Comment:
```suggestion
                "^(?<rank>[" + RANK_STRING + "]) §#[0-9a-f]{8}(?<name>[\\w ]{1,20})§. has just logged in!$");
```

We usually don't do that whole spec, just . it.

Reply:
discoverd `\p{XDigit}` for `[0-9a-fA-F]` maybe we can use that?

Reply:
Readability is still non existent when using that (which is the problem with the other one as well).